pg_search/src/postgres/storage/avl.rs

```diff
@@ -0,0 +1,1230 @@
+        // Use a sequence that approximates a balanced tree (e.g., insert in sorted order with some variation)
+        let keys = vec![
+            100, 50, 150, 25, 75, 125, 175, 12, 37, 62, 87, 112, 137, 162, 187, 6, 18, 31, 43, 56,
+            68, 81, 93, 106, 118, 131, 143, 156, 168, 181, 193,
+        ];
+        for &k in &keys {
+            let v = k * 2;
+            let result = m.insert(k, v);
+            assert!(result.is_ok(), "Insert failed for key {k}");
+            let (old_val, tag) = result.unwrap();
+            assert_eq!(old_val, None, "Unexpected old value for key {k}");
+            reference.insert(k, (v, tag));
+        }
+
+        // Verify iterator produces keys in order
+        let mut expected_keys = keys.clone();
+        expected_keys.sort();
+        let iter_keys: Vec<i32> = m.view().iter().map(|(k, _)| k).collect();
+        assert_eq!(
+            iter_keys, expected_keys,
+            "Iterator produced incorrect key order"
+        );
+
+        // Verify key/value/tag consistency
+        for (&k, &(v, t)) in &reference {
+            let got = m.view().get(&k);
+            assert_eq!(got, Some((v, t)), "Tag mismatch for key {k}");
+        }
+
+        // Delete some keys to trigger two-child deletions and slot reuse
+        let delete_keys = vec![100, 50, 150];
+        for &k in &delete_keys {
+            let removed = m.remove(&k);
+            let expected_val = reference.remove(&k).map(|(v, _)| v);
+            assert_eq!(removed, expected_val, "Remove failed for key {k}");
+        }
+
+        // Re-insert into freed slots
+        let new_keys = vec![90, 110, 130];
+        for &k in &new_keys {
+            let v = k * 2;
+            let result = m.insert(k, v);
+            assert!(result.is_ok(), "Insert failed for key {k}");
+            let (old_val, tag) = result.unwrap();
+            assert_eq!(old_val, None, "Unexpected old value for key {k}");
+            reference.insert(k, (v, tag));
+        }
+
+        // Verify iterator and tags again
+        let mut expected_keys: Vec<i32> = reference.keys().copied().collect();
+        expected_keys.sort();
+        let iter_keys: Vec<i32> = m.view().iter().map(|(k, _)| k).collect();
+        assert_eq!(
+            iter_keys, expected_keys,
+            "Iterator produced incorrect key order after deletions/insertions"
+        );
+
+        for (&k, &(v, t)) in &reference {
+            let got = m.view().get(&k);
+            assert_eq!(got, Some((v, t)), "Tag mismatch for key {k}");
+        }
+
+        // Verify tree height is reasonable (should be ~log2(32) ≈ 5-6)
+        let max_height = m.view().height(m.header.root);
+        assert!(
+            max_height <= 10,
+            "Tree height {max_height} is unexpectedly large"
+        );
+
+        #[cfg(debug_assertions)]
+        m.view().assert_ok();
+    }
+
+    proptest! {
+        #[test]
+        fn test_avl_map_prop(operations: Vec<(i32, i32)>) {
```

Comment:
I'm not sure what is happening here. I think that because this isn't `operations in ...` (note the `in`), this may not be running.

But I think that `test_avl_map_random_ops`/`test_avl_mapget_lte_matches_btreemap` are way more comprehensive anyway, so maybe remove this one?